Assets/Scripts/navigation/Navigation.cs

```diff
@@ -0,0 +1,186 @@
+		}
+		public void Setdestination(string target, int ID) {
+			if (target == null) {
+				Debug.LogError("Didn't find the city");
+				return;
+			}
+			Vector3 targetlocaion = cities.GetCityLocation(target);
+			if (targetlocaion == null) {
+				Debug.LogError("Unable to find city location");
+				return;
+			}
+			//assign image
+			ImageSlot.sprite = Resources.Load<Sprite>(ResourcePath + "/" + ID) ?? Resources.Load<Sprite>(DefaultPortrait);
+			StartNavigation = true;
+			//find player
+			player = GameObject.Find("playerShip").transform.Find("ShipParent").transform.Find("kyrenia_parent").transform;
+			this.transform.position = player.position;
+			float distance = Vector3.Distance(targetlocaion, player.position);
+			float radius = 2;
+			if (distance >= radius) {
+				string Saildirecation = FindPlayerDirection(player); // find what angle is player faceing
+				string Citydirecation = FindCityDirection(targetlocaion,player); // find city angle by the world
+				string WhereToSail = getAIPath(targetlocaion, radius); // find the path
+				text.text = Citydirecation +Saildirecation + WhereToSail; // assign text
+			}
+			if (distance < radius) {
+				text.text = "You have arrive to " + target + ".";
+			}
+		}
+		string FindCityDirection(Vector3 targetlocaion, Transform current) {
+			float angle = CalcAngle(current.position + -current.up, targetlocaion);    // target angle relative to world
+			angle -= 180;
+			string citydirecation = "The city is in the "+ FindDirection(angle)+".";
+			return citydirecation;
+		}
+		string FindPlayerDirection(Transform current) {
+			float angle2 = CalcAngle(current.position, current.position + current.up); // player angle relative to world
+			string Saildirecation = "The Ship is sail to the " + FindDirection(angle2) + ".";
+			return Saildirecation;
+		}
+		string getAIPath(Vector3 target,float radius) {
+			AI.stoppingDistance = radius;
+			AI.SetDestination(target);
+			Vector3[] path = AI.path.corners;
+			//lineRenderer.positionCount = AI.path.corners.Length;
+			if (path.Length <= 1) {
+				return "";
+			}
+			//for (int i = 0; i < path.Length; i++) {
+			//	lineRenderer.SetPosition(i, path[i]);
+			//}
+			//get next point
+			if(path.Length > 3) {
+				for (int i = 1; i < AI.path.corners.Length; i++) {
+					float dis = Vector3.Distance(player.position, AI.path.corners[i]);
+					if (dis < 6) {
+						int z = (i + 1) <= AI.path.corners.Length ? i + 1 : (AI.path.corners.Length - 1);
+						for (int j = i; j <= AI.path.corners.Length - 1; j++) {
+							var tmp = (j + 1) <= (AI.path.corners.Length - 1) ? Vector3.Distance(AI.path.corners[j], AI.path.corners[j + 1]) : Vector3.Distance(AI.path.corners[j], AI.path.corners[AI.path.corners.Length - 1]);
+							if (tmp > 5) {
+								z = (j + 1) <= AI.path.corners.Length ? j + 1 : (AI.path.corners.Length - 1);
+								break;
+							}
+						}
+						nexPoint = AI.path.corners[z];
+						break;
+					}
+					nexPoint = AI.path.corners[i];
+					break;
```

Comment:
you always call break here (it's not wrapped in an if like the break you have in the inner for loop), so the outer for loop on i always runs only once. should simplify that so there's no loop or see if maybe that's a logic bug that might cause issues. i noticed it because there's an "unreachable code detected" error on the i++ since the compiler can tell that you never get back to the top of the loop